extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/Limits.java

```diff
@@ -96,4 +96,11 @@ public class Limits
    * Max number of partition buckets for ingestion queries.
    */
   public static final int MAX_PARTITION_BUCKETS = 5_000;
+
+  /**
+   * Max number of rows with the same key in a window. This acts as a guardrail for
+   * data distribution with high cardinality
```

Comment:
is it for high cardinality or low cardinality?

Reply:
This is for the key with many unique values under it. Since this one key can have many possible values under it I'm calling it high cardinality